api/repositories/deployment_repository.go

```diff
@@ -0,0 +1,163 @@
+package repositories
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"code.cloudfoundry.org/korifi/api/authorization"
+	apierrors "code.cloudfoundry.org/korifi/api/errors"
+	korifiv1alpha1 "code.cloudfoundry.org/korifi/controllers/api/v1alpha1"
+	"code.cloudfoundry.org/korifi/controllers/controllers/workloads"
+	"code.cloudfoundry.org/korifi/tools/k8s"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const DeploymentResourceType = "Deployment"
+
+type DeploymentRepo struct {
+	userClientFactory  authorization.UserK8sClientFactory
+	namespaceRetriever NamespaceRetriever
+	rootNamespace      string
```

Comment:
What is this used for?

Reply:
We have removed the rootNamespace at it was unused, thanks for spotting